designs/plugins-phase-2.md

```diff
@@ -0,0 +1,105 @@
+
+Boilerplate is not currently in the plugin Universe because it’s language specific and depending on the language, it could change. Some alternatives to this are using a boilerplate path and the plugin could check for its existence, or it could exist in the universe not as a special file but as a list of strings without comments or markers which could be read line by line, thereby adding it to any file. Along with the above `plugin.Universe`, all the raw flags that would be passed to the subcommand need to be passed along across stdin/stdout. And the idea here is to make the plugin generic such that the args are passed as they are without specifying the resource explicitly, which eliminates the need to write a plugin struct to support every possible subcommand like init, create api, and create webhook. 
+
+
+## Configuration
+
+### Project file plugin type & enhanced `layout`
+
+Currently, the `PROJECT` file specifies what base plugin generated the project under a `layout` key (which is the plugin semantic version) in the format: `Plugin.Name() + "/" + Plugin.Version()`.
+
+To support phase 2 plugins, we propose that we modify the "layout" PROJECT file key to either of these options as order matters for certain plugins: 
+    * Ordered list of plugin {name}/{version} or
+    * Ordered map of {name}/{version} to plugin-specific config
+
+Example `PROJECT` file:
+
+```yaml
+version: "3-alpha"
+layout: ["go/v1.0.0", "go.kubebuilder.io/v3-alpha"]
```

Comment:
the layout means the father of the plugins used to scaffold the project. So, what would mean `"go/v1.0.0", "go.kubebuilder.io/v3-alpha"`? Could we give an example of a scenario that we would like to address and then have the project file as it would be in this scenario for a better understand?

Reply:
I think this example is confusing, as both plugins are the base go plugin (one with the short form and the other one the full name) which doesn't make sense. Additionally, we have `v2` and `v3` versions of that plugin, no `v1.0.0` nor `v3-alpha`, to add up to the confussion.